src/graphql/context.ts

```diff
@@ -8,13 +9,35 @@ export interface IContext {
 }
 
 export class Context implements IContext {
-	private accountId: string;
+	private signedMessage: string;
 
-	constructor(accountId: string) {
-		this.accountId = accountId;
+	/**This constructor will recieve singed messages for any transactions
+	 * and return the address which the message came from to be used as
+	 * context variable in other scripts.
+	 */
+	constructor(signedMessage: string) {
+		this.signedMessage = signedMessage;
 	}
 
 	public getEtheriumAccountId(): string {
-		return this.accountId;
+		return this.findAccount();
+	}
+
+	private findAccount() {
+		const web3 = new Web3(
+			new Web3.providers.HttpProvider('http://localhost:7545')
+		);
+
+		let etheriumAccountId: string = '';
+
+		try {
+			etheriumAccountId = web3.eth.accounts.recover(

```

Comment:
Lets do some checking here to make sure the return value is a non empty string and then return it, and if not, return null. Also report all these errors with console.error just so we know.